### PR TITLE
feat: add `BabyBear4` C api

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR adds `BabyBear4` C api for Plonky3 binding.

### CODE REVIEW

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: tachyon/math/finite_fields/generator/ext_field_generator/build_defs.bzl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 load("//bazel:tachyon_cc.bzl", "tachyon_cc_library")
 
 def _generate_ext_field_impl(ctx):
-    fq_hdr_tpl_path = ctx.expand_location("$(location @kroma_network_tachyon//tachyon/math/finite_fields/generator/ext_field_generator:fq.h.tpl)", [ctx.attr.fq_hdr_tpl_path])
+    fq_hdr_tpl_path = ctx.expand_location("$(location @kroma_network_tachyon//tachyon/math/finite_fields/generator/ext_field_generator:fp.h.tpl)", [ctx.attr.fq_hdr_tpl_path])
```
Comment (ashjeong) [inline tachyon/math/finite_fields/generator/ext_field_generator/build_defs.bzl:4]: + fix for all instances of `fq_hdr_tpl_path`
```suggestion
    fp_hdr_tpl_path = ctx.expand_location("$(location @kroma_network_tachyon//tachyon/math/finite_fields/generator/ext_field_generator:fp.h.tpl)", [ctx.attr.fp_hdr_tpl_path])

File: tachyon/c/math/finite_fields/baby_bear/BUILD.bazel [hunk 2]
```diff
@@ -7,14 +7,11 @@ filegroup(
     srcs = ["baby_bear.h"],
 )
 
-tachyon_cc_library(
+generate_small_prime_fields(
```
Comment (ashjeong) [inline tachyon/c/math/finite_fields/baby_bear/BUILD.bazel:14]: Random question, but don't these bazel targets only general one field at a time? So technically it should be `generate_small_prime_field`?

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

File: tachyon/c/math/finite_fields/baby_bear/BUILD.bazel (hunk 2 above)
Comment (chokobole) [reply tachyon/c/math/finite_fields/baby_bear/BUILD.bazel:14]: From the perspective of the field, as you said, it's one. But from the perspective of the build targets(files), it's many e.g, The build targets that produce `baby_bear.h`, `baby_bear.cc`, and `baby_bear_types_traits.h` respectively.

**Review by batzor — APPROVED** [review]
LGTM

**Review by GideokKim — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**